client/platform/desktop/frontend/components/Settings.vue

```diff
@@ -33,8 +33,13 @@ export default defineComponent({
     const pendingChanges = computed(() => isEqual(localSettings.value, settings.value));
 
     onBeforeMount(async () => {
-      settingsAreValid.value = await validateSettings(localSettings.value);
       smi.value = await nvidiaSmi();
+      settingsAreValid.value = await validateSettings(localSettings.value);
```

Comment:
You flipped the order here.  Does this change anything?  If this order matters, you have a race condition that needs to be addressed.